include/tvm/tir/builtin.h

```diff
@@ -92,6 +92,14 @@ TVM_DLL const Op& shift_right();
  */
 TVM_DLL const Op& large_uint_imm();
 
+/*!
+ * \brief Execute a multiplication between two Q-numbers x and y
+ * followed by a right shift s
+ * The default rounding rule is to the nearest value, rounding half up
+ * (i.e., round(x.1) = x and round (x.5) = x+1)
+ */
+TVM_DLL const Op& qmuls();
```

Comment:
We should come up with a better name. Currently, `qmuls` seems vague.
Not sure what `q` and `s` stand for a person not familiar with Q numbers.

Why not use the same `fixed_point_multiply`?

Reply:
Should we use this thread to discuss naming? The problem with using `fixed_point_multipliy` for the intrinsic is that it describes an operation which is more general than the `fixed_point_multiply` we do in Relay. So `fixed_point_multiply(x,m,s) = qmuls(x, m, 31, s)`. We might call it `q_multiply_shift`? Or `q_multiply_scale`?

Reply:
`q_multiply_shift` is better imo compared to `qmuls`

@kparzysz-quic Any suggestions here?

Reply:
Same here, I like `q_multiply_shift`